web/__init__.py

```diff
@@ -32,60 +52,77 @@
-    app.svc.register("filetransfer", web.service.filetransfer.FileTransferService())
+    if app.config["login"]:
+        app.svc.register("pppp", web.service.pppp.PPPPService())
+        app.svc.register("videoqueue", web.service.video.VideoQueue())
+        app.svc.register("mqttqueue", web.service.mqtt.MqttQueue())
+        app.svc.register("filetransfer", web.service.filetransfer.FileTransferService())
 
 
 @sock.route("/ws/mqtt")
 def mqtt(sock):
 
-    for data in app.svc.stream("mqttqueue"):
-        log.debug(f"MQTT message: {data}")
-        sock.send(json.dumps(data))
+    if app.config["login"]:
+        for data in app.svc.stream("mqttqueue"):
+            log.debug(f"MQTT message: {data}")
+            sock.send(json.dumps(data))
 
 
 @sock.route("/ws/video")
 def video(sock):
 
-    for msg in app.svc.stream("videoqueue"):
-        sock.send(msg.data)
+    if app.config["login"]:
+        for msg in app.svc.stream("videoqueue"):
+            sock.send(msg.data)
 
 
 @sock.route("/ws/ctrl")
 def ctrl(sock):
 
-    while True:
-        msg = json.loads(sock.receive())
+    if app.config["login"]:
+        while True:
+            msg = json.loads(sock.receive())
 
-        if "light" in msg:
-            with app.svc.borrow("videoqueue") as vq:
-                vq.api_light_state(msg["light"])
+            if "light" in msg:
+                with app.svc.borrow("videoqueue") as vq:
+                    vq.api_light_state(msg["light"])
 
-        if "quality" in msg:
-            with app.svc.borrow("videoqueue") as vq:
-                vq.api_video_mode(msg["quality"])
+            if "quality" in msg:
+                with app.svc.borrow("videoqueue") as vq:
+                    vq.api_video_mode(msg["quality"])
 
 
 @app.get("/video")
 def video_download():
 
     def generate():
-        for msg in app.svc.stream("videoqueue"):
-            yield msg.data
+        if app.config["login"]:
+            for msg in app.svc.stream("videoqueue"):
+                yield msg.data
 
     return Response(generate(), mimetype='video/mp4')
 
 
 @app.get("/")
 def app_root():
+    config = app.config["config"]
+    user_agent = parse(request.headers.get('User-Agent'))
+    login_path = {
+        'Mac OS': '~/Library/Application Support/AnkerMake/AnkerMake_64bit_fp/login.json',
+        'Windows': r'%LOCALAPPDATA%\Ankermake\AnkerMake_64bit_fp\login.json',
+        'None': 'Unsupported OS, supply path to login.json',
+    }
+    useros = dict_match(login_path, user_agent.os.family)
+
     host = request.host.split(':')
     requestPort = host[1] if len(host) > 1 else '80' # If there is no 2nd array entry, the request port is 80
     return render_template(
         "index.html",
         requestPort=requestPort,
-        requestHost=host[0]
+        requestHost=host[0],
+        configure=app.config["login"],
+        loginFilePath=login_path[useros] if useros in login_path else login_path["None"],
```

Comment:
I know you're only following the naming scheme here, but it is unfortunately inconsistent with everything else.

Please change to `snake_case` instead of `smallCamelCase`.

Reply:
Also, what is `login_path["None"]`? Is it an error? Seems confusing.

Reply:
Lines 110-114: 

```python
    login_path = {
        'Mac OS': '~/Library/Application Support/AnkerMake/AnkerMake_64bit_fp/login.json',
        'Windows': r'%LOCALAPPDATA%\Ankermake\AnkerMake_64bit_fp\login.json',
        'None': 'Unsupported OS, supply path to login.json',
    }
```

Reply:
Regarding case, I will go back and change all of the new variables to snake case instead of Camel Case.